packages/x-components/src/components/result/__tests__/result-provider.spec.ts

```diff
@@ -0,0 +1,247 @@
+  const [, localVue] = installNewXPlugin();
+
+  /**
+   * Component to test the provided items of the provider.
+   * It exposes these items in the default slot to facilitate the tests.
+   */
+  @Component({
+    template: `
+      <div>
+        <slot
+          :setResultVariant="setResultVariant"
+          :originalResult="originalResult"
+          :selectedVariants="selectedVariants"
+        ></slot>
+      </div>
+    `
+  })
+  class Child extends Vue {
+    @XInject(SET_RESULT_VARIANT_KEY)
+    public setResultVariant!: () => void;
+
+    @XInject(RESULT_KEY)
+    public originalResult!: Result;
+
+    @XInject(SELECTED_VARIANTS_KEY)
+    public selectedVariants!: ResultVariant[];
+
+    @Prop()
+    public result!: Result;
+  }
+
+  const wrapper = mount(
+    {
+      template: `<ResultProvider :result="result" #default="{ result }">
+          ${template}
+        </ResultProvider>`,
+      components: {
+        ResultProvider,
+        Child
+      }
+    },
+    {
+      localVue,
+      props: ['result'],
+      propsData: {
+        result
+      },
+      scopedSlots: {
+        default: () => {
+          return template;
+        }
+      }
+    }
+  );
+
+  return {
+    wrapper: wrapper.findComponent(ResultProvider),
+    result
+  };
+};
+
+describe('result with variants provider', () => {
+  it('exposes the result in the default slot', () => {
+    const template = `
+      <span data-test="result-name">{{result.name}}</span>
+    `;
+    const result = createResultStub('tshirt');
+    const { wrapper } = renderResultProvider({ result, template });
+    expect(wrapper.find(getDataTestSelector('result-name')).text()).toBe('tshirt');
+  });
+
+  it('provides a callback to set the selected variant', async () => {
+    const { wrapper } = renderResultProvider({
+      template: `
+        <Child :result="result" #default="{setResultVariant}">
+          <button
+              data-test="set-red-jacket"
+              @click="setResultVariant(0, result.variants[0])">
+            Change variant
+          </button>
+          <button
+              data-test="set-red-xl-jacket"
+              @click="setResultVariant(1, result.variants[0].variants[1])">
+            Change variant
+          </button>
+          <span data-test="result-name">{{ result.name }}</span>
+        </Child>
+      `
+    });
+    const setRedJacketButton = wrapper.find(getDataTestSelector('set-red-jacket'));
+    const setRedJacketXlButton = wrapper.find(getDataTestSelector('set-red-xl-jacket'));
+
+    expect(wrapper.find(getDataTestSelector('result-name')).text()).toBe('jacket');
+
+    await setRedJacketButton.trigger('click');
+
+    expect(wrapper.find(getDataTestSelector('result-name')).text()).toBe('red jacket');
+
+    await setRedJacketXlButton.trigger('click');
+
+    expect(wrapper.find(getDataTestSelector('result-name')).text()).toBe('red jacket L');
+
+    // It won't deselect the child variant if the parent is clicked.
```

Comment:
Yep, it will be implemented in that task.
But this behavior will depend on the client, in Oysho for example the first color should be selected by default, but not the first size.